cmd/atlas/internal/url/url.go

```diff
@@ -0,0 +1,85 @@
+package url
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"ariga.io/atlas/sql/schema"
+)
+
+type (
+	providerMap struct {
+		m map[string]func(string) (*Atlas, error)
+	}
+
+	// Mux is used for routing dsn to correct provider.
+	Mux struct {
+		providers *providerMap
+	}
+
+	// Atlas implements the Driver interface using Atlas.
+	Atlas struct {
+		DB        *sql.DB
+		Differ    schema.Differ
+		Execer    schema.Execer
+		Inspector schema.Inspector
+	}
+)
+
+// Close release all resources taken by Driver.
+func (a *Atlas) Close() error {
+	return a.DB.Close()
+}
+
+// NewMux returns a new Mux.
+func NewMux() *Mux {
+	return &Mux{
+		providers: &providerMap{
+			m: make(map[string]func(string) (*Atlas, error)),
+		},
+	}
+}
+
+var defaultURLMux *Mux
+
+// DefaultURLMux returns the default system "Mux".
+func DefaultURLMux() *Mux {
+	if defaultURLMux == nil {
+		defaultURLMux = NewMux()
+	}
+	return defaultURLMux
+}
```

Comment:
```suggestion
var defaultURLMux = NewMux()

// DefaultURLMux returns the default system "Mux".
func DefaultURLMux() *Mux {
	return defaultURLMux
}
```

Why do you need the `DefaultURLMux` func? Also, decide if it's `URLMux` or just `Mux`.

Reply:
Mux,
This pattern is used by the go-cloud package, I think it's to avoid using a var straight from the package.